plugin/trino-hive/src/main/java/io/trino/plugin/hive/InternalHiveSplit.java

```diff
@@ -75,7 +79,7 @@ public InternalHiveSplit(
             List<HivePartitionKey> partitionKeys,
             List<InternalHiveBlock> blocks,
             OptionalInt bucketNumber,
-            int statementId,
+            Function<Integer, Integer> statementIdSupplier,
```

Comment:
This could be a supplier, since the InternalHiveSplit class is for a single bucket.

Reply:
Good point; changed.